File: task1/ruby/main.rb
def min_operations(s)
  n = s.length
  max_i = -1
  # Перебираем возможные длины подстроки, начиная с наибольшей
  (n / 2 - 1).downto(0) do |i|
    substr_len = i + 1
    next if 2 * substr_len > n
    # Проверяем, совпадает ли первая подстрока с последующей
    if s[0...substr_len] == s[substr_len...(2 * substr_len)]
      max_i = i
      break
    end
  end
  # Возвращаем минимальное количество операций
  max_i != -1 ? n - max_i : n
end

#тут получаем строку от пользователя, обрабатываем, выводим
puts "Введите строку:"
s = gets.chomp
puts "Результат: #{min_operations(s)}"

File: task1/java/main.java
import java.util.Scanner;

public class main {
    public static int minOperations(String s) {
        int n = s.length();
        int maxI = -1;
        // Перебираем возможные длины подстроки, начиная с наибольшей
        for (int i = n / 2 - 1; i >= 0; i--) {
            int substrLen = i + 1;
            if (2 * substrLen > n) {
                continue;
            }
            // Проверяем, совпадает ли первая подстрока с последующей
            if (s.substring(0, substrLen).equals(s.substring(substrLen, 2 * substrLen))) {
                maxI = i;
                break;
            }
        }
        // Возвращаем минимальное количество операций
        return maxI != -1 ? n - maxI : n;
    }

    public static void main(String[] args) {
        //тут получаем строку от пользователя, обрабатываем, выводим
        Scanner scanner = new Scanner(System.in);
        System.out.print("Введите строку: ");
        String s = scanner.nextLine().trim();
        System.out.println("Результат: " + minOperations(s));
        scanner.close();
    }
}
